src/main/java/pl/betse/beontime/timesheet/controller/WeekTimeEntryController.java

```diff
@@ -0,0 +1,109 @@
+package pl.betse.beontime.timesheet.controller;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.hateoas.Link;
+import org.springframework.hateoas.Resources;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+import pl.betse.beontime.timesheet.bo.TimeEntryBo;
+import pl.betse.beontime.timesheet.exception.IncorrectWeekFormatException;
+import pl.betse.beontime.timesheet.mapper.TimeEntryMapper;
+import pl.betse.beontime.timesheet.model.WeekTimeEntryBody;
+import pl.betse.beontime.timesheet.service.TimeEntryService;
+import pl.betse.beontime.timesheet.validation.CreateTimeEntry;
+
+import javax.servlet.http.HttpServletRequest;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.springframework.hateoas.mvc.ControllerLinkBuilder.linkTo;
+import static org.springframework.hateoas.mvc.ControllerLinkBuilder.methodOn;
+
+@Slf4j
+@RestController
+@RequestMapping("/consultants")
+public class WeekTimeEntryController {
+    private final TimeEntryService timeEntryService;
+    private final TimeEntryMapper timeEntryMapper;
+
+    public WeekTimeEntryController(TimeEntryService timeEntryService, TimeEntryMapper timeEntryMapper) {
+        this.timeEntryService = timeEntryService;
+        this.timeEntryMapper = timeEntryMapper;
+    }
+
+    @GetMapping("/{userGuid}/week/{weekNumber}")
+    public ResponseEntity<Resources<WeekTimeEntryBody>> getWeekForUser(
+            @PathVariable("userGuid") String userGuid,
+            @PathVariable("weekNumber") String weekNumber) {
+        checkWeekNumberFormat(weekNumber);
+        List<List<TimeEntryBo>> timeEntryBoList = timeEntryService.findByUserGuidAndWeek(userGuid, weekNumber);
+        List<WeekTimeEntryBody> weekTimeEntryBodyList = new ArrayList<>();
+        timeEntryBoList.forEach(week ->
+                weekTimeEntryBodyList.add(timeEntryMapper.fromTimeEntryBoToWeekTimeEntryBody(week))
+        );
+        URI location = linkTo(methodOn(WeekTimeEntryController.class).getWeekForUser(userGuid, weekNumber)).toUri();
+        Link link = new Link(location.toString(), "self");
```

Comment:
Link should be part of the Body object, not the resources one  - now you will get only one link for whole collection, which is not correct

Reply:
We treat whole this endpoint collection as a single result of single request, because all CRUD operations for user (person who will fill the timesheet form in FE app) will be operated weekly. Other operations provided by manager (person who will approve or reject users entries) will be operated monthly.